### PR TITLE
tests: verify all deployments are ready after installation

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Makes E2E tests more restrictive about deployments' readiness before proceeding with configuring any resources. It's to ensure that both Gateway and KIC deployments are ready in a default idle state after installation. This is the behavior we want to preserve while implementing #3499.

**Which issue this PR fixes**:

Part of #3499.

### CODE REVIEW

File: test/e2e/utils_test.go [hunk 1]
```diff
@@ -489,3 +492,14 @@ func setEnv(p setEnvParams) error {
 	}
 	return nil
 }
+
+// dumpToTempFile dumps the contents of the reader to a temporary file and returns the path to the file.
+func dumpToTempFile(t *testing.T, reader io.Reader) string {
+	t.Helper()
+	file, err := os.CreateTemp("", "")
+	require.NoError(t, err)
+	defer file.Close()
```
Comment (pmalek) [inline test/e2e/utils_test.go:501]: Why not use `t.TempDir()`? It automatically removes the file during `t.Cleanup()`

File: test/e2e/utils_test.go [hunk 2]
```diff
@@ -108,7 +108,7 @@ func exposeAdminAPI(ctx context.Context, t *testing.T, env environments.Environm
 
 // getTestManifest gets a manifest io.Reader, applying optional patches to the base manifest provided.
 // In case of any failure while patching, the base manifest is returned.
-func getTestManifest(t *testing.T, baseManifestPath string) io.Reader {
+func getTestManifest(t *testing.T, baseManifestPath string, skipTestPatches bool) io.Reader {
```
Comment (pmalek) [inline test/e2e/utils_test.go:111]: Can we add a godoc for `skipTestPatches` parameter?

File: test/e2e/utils_test.go [hunk 3]
```diff
@@ -118,38 +119,41 @@ func getTestManifest(t *testing.T, baseManifestPath string) io.Reader {
...
-	}
+	if !skipTestPatches {
+		t.Logf("applying test patches to manifest %v", baseManifestPath)
 
-	manifestsReader, err = patchGatewayImageFromEnv(t, manifestsReader)
-	if err != nil {
-		t.Logf("failed patching gateway image (%v), using default manifest %v", err, baseManifestPath)
-		return manifestsReader
-	}
+		manifestsReader, err = patchControllerImageFromEnv(t, manifestsReader)
```
Comment (randmonkey) [inline test/e2e/utils_test.go:125]: Are the spaces redundant?

Comment (czeslavo) [reply test/e2e/utils_test.go:125]: I don't think so, it's an indentation that was introduced for `if !skipTestPatches {}` block.

**Review by randmonkey — APPROVED** [review]

**Pull request merged**